### ISSUE TITLE
[JavaScript] Implement new Concept Exercise: concept/arrays-analysis

### ISSUE THREAD

**Issue by peterchu999** [exercism/v3#1370]
This issue describes how to implement the arrays-analysis concept exercise for the JavaScript track and is a continuation of #1300 

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

the goals of this exercise is to teach student how to analyse /check the value within the array without looping with javascript

## Learning objectives

- understand the use of  `includes`
- understand the use of  `every`
- understand the use of  `some`
- understand the use of  `find`
- understand the use of  `indexOf`

## Out of scope

- the usage of looping function like (map, forEach, filter, and reduce).
- the usage of `sort` in array
- the usage of `reverse` in array
- the usage of `flat` to make a nested array to become 1 level

## Concepts

- learn to `analyse` / check value in array without looping through the array
- learn how to locate position of an value within array

## Prerequisites

- `number`
- `boolean`
- `arrays`
- `function`

### Hints

- [MDN `array` Documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
)

### After

- [MDN `array` Documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
)

## Representer

No changes needed.

## Analyzer

No changes needed.

## Implementing

See the [`Implementation Guide for JavaScript Concept Exercises`](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise)  for more details.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by SleeplessByte** [comment]
Very very nice. I'm excited about this. Please post if you have an idea for the exercise itself?

**Comment by peterchu999** [comment]
i think i can make it by modify the story a little bit. i'll fork and make the pull req when it's done
